### DISCUSSION TITLE
Convert hurl script to curl command

### DISCUSSION THREAD

**Discussion by GlassGruber** [Orange-OpenSource/hurl#2615] [Q&A] [answered]
Hey there, hurl it's amazing!

I have a couple of scripts, and it happened that I needed to run commands on a remote machine where hurl is not present. 
Is there a command that allows me to translate a hurl script for easy copy paste for curl?

Test/asserts are not necessary, jsut quick way to translate a script with command or chains would be really cool for an old lazy bum like me 🙃 

I hope this is not a RTFM issue, I swear I've checked! :P

**Comment by jcamiel** [comment, marked as answer]
😆if it's in the docs, it's well buried!

You can go from curl to Hurl with `hurlfmt`. To go the other way, you can run your Hurl file with `--verbose` option and grep the output to see each request's curl equivalent command. Dynamic values (variables/captures) are resolved so you should be able to replay your Hurl file.

**Reply by GlassGruber** [reply]
You are indeed absolutely right, also truth be told it is sort of mentioned in the docs in the [Debug Tips](https://hurl.dev/docs/tutorial/debug-tips.html) page. Verbose mode is very well formatted my hat to you and the team! 🎩 👌

**Reply by jcamiel** [reply]
I'll update the page to emphasise that verbose logs output curl commands.

**Comment by GlassGruber** [comment]
A slightly OT follow up question: I guess is not available since it is not available in [curl as well](https://github.com/curl/curl/issues/5426), is it possible to do a sort of dry run or preview of command before executing it? 

I think this is kinda related to current thread since, if I have to run the command in verbose mode to obtain the curl command, maybe it is not appropriate to execute again that command for whatever reason.

Thank you again for your answer!

**Reply by jcamiel** [reply]
Yes, it's not possible because some request parameters are dynamic and can depends on previous requests (even previous response data!). Every request data can be [variabilized](https://hurl.dev/docs/templates.html) so it's not possible to create the curl commands without running the requests!

**Reply by GlassGruber** [reply]
I see, I guess the most sensed way is changing the target to a null server or something like that. 
Thank you for your time!
